File: lib/gcd.py
"""
this program will find the greatest common divisor of two
positive integers provided by the user using Euclid's algorithm
"""

s = int(input('Enter the first smaller number: '))
while s < 1:
    s = int(input('\tNumbers should be positive. Try again: '))

b = int(input('Enter the second larger number: '))
while b < 1:
    s = int(input('\tNumbers should be positive. Try again: '))
# error check to make sure the second number is larger than the first
while b < s:
    print('\tThe second number needs to be larger than the first, try again.')
    s = int(input('Enter the first smaller number: '))
    while s < 1:
        s = int(input('Numbers should be positive: '))

    b = int(input('Enter the second larger number: '))
    while b < 1:
        s = int(input('Numbers should be positive: '))
# Euclid's algorithm:
while s != 0:
    r = b % s
    b = s
    s = r
print()
print('The greatest common divisor is', b)

File: lib/gcd.rb
'''
GCD class finds the greatest common divisor of two numbers, num1 and num2
with ValueError if num1 is not positive
or if num2 is smaller than num1
'''
class GCD
  def initialize(num1, num2)
    @num1 = num1
    @num2 = num2
  end

  def num1=(num1)
    if num1 < 0
      raise ValueError
    end
  end

  def num2=(num2)
    if num2 < @num1
      raise ValueError
    end
  end

  def find_gcd
    # find greatest common divisor for num1 and num2
    while @num1 != 0
      r = @num2 % @num1
      @num2 = @num1
      @num1 = r
    end
    @num2
  end
end
